sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/channel/AnalyticsValidator.java

```diff
@@ -127,4 +139,69 @@ private static String validateName(String name, String logType) {
         }
         return result;
     }
+
+    /**
+     * Validates typed properties.
+     *
+     * @param properties Typed properties collection to validate.
+     */
+    private static void validateProperties(List<TypedProperty> properties) {
+        if (properties == null) {
+            return;
+        }
+        int count = 0;
+        boolean maxCountReached = false;
+        String message;
+        for (Iterator<TypedProperty> iterator = properties.iterator(); iterator.hasNext(); ) {
+            TypedProperty property = iterator.next();
+            String key = property.getName();
+            if (count >= MAX_PROPERTY_COUNT) {
+                if (!maxCountReached) {
+                    message = String.format("Typed properties cannot contain more than %s items. Skipping other properties.", MAX_PROPERTY_COUNT);
+                    AppCenterLog.warn(LOG_TAG, message);
+                    maxCountReached = true;
+                }
+                iterator.remove();
+                continue;
+            }
+            if (key == null || key.isEmpty()) {
+                AppCenterLog.warn(LOG_TAG, "A typed property key cannot be null or empty. Property will be skipped.");
+                iterator.remove();
+                continue;
+            }
+            if (key.length() > MAX_PROPERTY_ITEM_LENGTH) {
+                message = String.format("Typed property '%s' : property key length cannot be longer than %s characters. Property key will be truncated.", key, MAX_PROPERTY_ITEM_LENGTH);
+                AppCenterLog.warn(LOG_TAG, message);
+                property.setName(key.substring(0, MAX_PROPERTY_ITEM_LENGTH));
```

Comment:
This should change a copy of the property, not the property itself. Even though they are moved to another array, that array contains references to properties that will be used in other logs as well. (same comment on line 189)